datalab_cohorts/__init__.py

```diff
@@ -426,16 +456,47 @@ def flatten_nested_covariates(self, covariate_definitions):
                 flattened[name] = (query_type, query_args)
         return flattened, hidden
 
-    def default_for_column(self, column_name):
-        is_str_col = (
-            column_name == "date"
-            or column_name.startswith("date_")
-            or column_name.endswith("_date")
-            or column_name.endswith("_code")
-            or column_name == "category"
-            or column_name.endswith("_name")
-        )
-        return "" if is_str_col else 0
+    def add_include_date_flags_to_columns(self, covariate_definitions):
+        """
+        Where one column is defined as being the date generated by another
+        column we need to tell that "source" column that it should calculate a
+        date as well, which we do by supplying a flag.
+
+        The sharp-eyed may notice that this is close to being the inverse of
+        `apply_compatibility_fixes` above and wonder what is being acheived.
+        But this gives us a way of incrementally refactoring things without
+        having to make big sweeping changes and without breaking backwards
+        compatibility.
+        """
+        updated = copy.deepcopy(covariate_definitions)
+        for name, (query_type, query_args) in updated.items():
+            if query_type == "value_from" and query_args["returning"] == "date":
```

Comment:
Do we need an assertion here that `date` is the only supported `returning` value?

Reply:
At the moment, `value_from` isn't part of the public API, you can only get to it via `date_of`. But you're right, an extra assert can't hurt.